code/game/objects/structures/pit.dm

```diff
@@ -175,9 +175,9 @@
 		if(W.do_tool_interaction(TOOL_HATCHET, user, src, 3 SECONDS, "hacking away at", "hacking at"))
 			physically_destroyed(FALSE)
 		return TRUE
-	if(istype(W,/obj/item/pen))
+	if(IS_PEN(W) && CanPhysicallyInteractWith(user, src))
```

Comment:
You only need `CanPhysicallyInteractWith` if you're specifying something other than `src` as the target. `CanPhysicallyInteract` will do the job here.

Reply:
I actually also goofed that one. The interaction check was meant to be after the player input. But noted!